### ISSUE TITLE
Trying to debug results for me in the following exception

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode#57945]
```stack
TypeError: Cannot convert object to primitive value
    at String.replace (<anonymous>)
    at t (c:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app/out/vs/loader.js:34:302)
    at Object.r._detected [as localize] (c:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app/out/vs/loader.js:34:523)
    at file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:4772:56
    at Object.g [as _notify] (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:154:729)
    at Object.enter (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:158:61)
    at n.Class.derive._creator._run (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:159:902)
    at n.Class.derive._creator._completed (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:159:343)
    at Object.g [as _notify] (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:154:729)
    at Object.enter (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:158:61)
    at n.Class.derive._oncancel._run (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:159:902)
    at n.Class.derive._oncancel._completed (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:159:343)
    at file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:4773:352
    at Object.g [as _notify] (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:154:729)
    at Object.enter (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:158:61)
    at n.Class.derive._creator._run (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:159:902)
    at n.Class.derive._creator._completed (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:159:343)
    at Object.g [as _notify] (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:154:729)
    at Object.enter (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:158:61)
    at n.Class.derive._oncancel._run (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:159:902)
    at n.Class.derive._oncancel._completed (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:159:343)
    at file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:4773:276
    at file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:4772:745
    at e.fire (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:175:76)
    at file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:4070:721
    at e.fire (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:175:76)
    at file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:3084:168
    at e.fire (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:175:76)
    at t.tryFinish (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:3058:498)
    at t.processLine (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:3057:279)
    at file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:3084:795
    at e.fire (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:175:76)
    at t._sendLineData (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:3516:317)
    at t._onLineFeed (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:3516:108)
    at Terminal.<anonymous> (file:///C:/Users/dirkb/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/workbench/workbench.main.js:3503:910)
    at Terminal.EventEmitter.emit (C:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\node_modules.asar\vscode-xterm\lib\EventEmitter.js:66:20)
    at InputHandler.lineFeed (C:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\node_modules.asar\vscode-xterm\lib\InputHandler.js:307:24)
    at C:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\node_modules.asar\vscode-xterm\lib\InputHandler.js:128:93
    at EscapeSequenceParser.parse (C:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\node_modules.asar\vscode-xterm\lib\EscapeSequenceParser.js:307:25)
    at InputHandler.parse (C:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\node_modules.asar\vscode-xterm\lib\InputHandler.js:196:22)
    at Terminal._innerWrite (C:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\node_modules.asar\vscode-xterm\lib\Terminal.js:871:32)
    at C:\Users\dirkb\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\node_modules.asar\vscode-xterm\lib\Terminal.js:853:23
```

**Comment by dbaeumer** [comment]
I was not able to reproduce this but saw it yesterday quite some time.

**Comment by weinand** [comment]
We've analysed the problem and it is a result of two problems:
- in debug-land we did not correctly adopt the change that a `preLaunchTask` is now either a string or an object literal: in https://github.com/Microsoft/vscode/blob/fa30debe4834b619228bb4e95d3c8295180e8dee/src/vs/workbench/parts/debug/electron-browser/debugService.ts#L680 the `preLaunchTask` property of a debug config is passed as an argument to the localize function which only expects strings.
- in the `localize` function `toString()` is called on the passed argument without checking that it has that method. And the launch config has no toString method because it is really a proxy object returned from the configuration service.
